Updates on VPC PSC forwarding rule

gemmahou opened the pull request:
### Change description

<!--
Describe what this pull request does.

* If your pull request is to address an open issue, indicate it by specifying the
issue number: 

For example: "Fixes #858"
-->

Follow up on PR#1942

Currently, only regional compute forwarding rule supports target service attachment. Global forwarding rule does not support this feature. Make our samples and fixtures yaml specific to regional forwarding rule to avoid confusion.

More details: https://b.corp.google.com/issues/342233672#comment6

### Tests you have done

<!--

Make sure you have run "make ready-pr" to run required tests and ensure this PR is ready to review. 

Also if possible, share a bit more on the tests you have done. 

For example if you have updated the pubsubtopic sample, you can share the test logs from running the test case locally.

go test -v -tags=integration ./config/tests/samples/create -test.run TestAll -run-tests pubsubtopic

-->

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

maqiuyujoyce commented on pkg/test/constants/constants.go:
```diff
@@ -45,7 +45,7 @@ var TestNameRegexesToSkip = []string{
 	".*(containerattachedcluster).*",
 	// Disable due to TF bug https://github.com/hashicorp/terraform-provider-google/issues/16255.
 	// We can't specify labels in the create operation, that causes AssertLabelsMatchAndHaveManagedLabel check to fail.
-	".*(privateserviceconnectforwardingrule).*",
```
Could you help me understand why do we want to make this change?

gemmahou replied:
privateserviceconnectforwardingrule is renamed to regionalforwardingrulepsc

maqiuyujoyce replied:
Gotcha! I got confused because I found the sample is renamed to `regional-forwardingrule-vpc-psc` but didn't realize this is the name for testdata.

maqiuyujoyce reviewed (commented):
/lgtm
/approve

/hold
for a quick question.

Meanwhile, as I may have the question, maybe we should document the context somewhere for other contributors?

maqiuyujoyce reviewed (commented):
/hold cancel

gemmahou commented:
> /lgtm /approve
> 
> /hold for a quick question.
> 
> Meanwhile, as I may have the question, maybe we should document the context somewhere for other contributors?

I added a comment [here](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/8028bb1eb832b180eaa3cd3aacf8b80b3400d107/config/samples/resources/computeforwardingrule/regional-forwarding-rule-vpc-psc/compute_v1beta1_computeforwardingrule.yaml#L40), also added a note in b/342233672. Since our resource docs are generated, I can log a ticket/open an issue to service team/TF to update their docs.
